src/components/common/ProductBox/ProductBox.js

```diff
@@ -45,11 +47,13 @@ const ProductBox = ({
     event.preventDefault();
     addToFavourite(id);
   };
-
+  let url = '/product/' + { id };
```

Comment:
Zapisując wyrażenie `{ id }` wykorzystałaś opcję wprowadzoną wraz ze standardem EcmaScript2015 do stworzenia obiektu korzystając ze skrótowego zapisu właściwości . W efekcie to wyrażenie równoważane jest takiemu zapisowi
```
{
  id: id
}
```

Do poczytania https://alligator.io/js/object-property-shorthand-es6/

Wystarczyło by to zapisać tak `'/product/' + id;` i możesz to bezpośrednio zapisać w komponenecie `<Link>`

Możesz też wykorzystać template literals do zapisania adresu https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals